### ISSUE TITLE
Joining a CS 1.6 server with CS CZ client causes STEAM validation rejected

### ISSUE THREAD

**Issue by SmilexGamer** [dreamstalker/rehlds#996]

**Comment by StevenKal** [comment]
Did you tried on your client to type: setinfo _gd cstrike
From [this](https://github.com/dreamstalker/rehlds/commit/0c5ce53666811c4243ed5aa0c755dab26d2283d4) change.

**Comment by SmilexGamer** [comment]
> Did you tried on your client to type: setinfo _gd cstrike From [this](https://github.com/dreamstalker/rehlds/commit/0c5ce53666811c4243ed5aa0c755dab26d2283d4) change, and by using [latest ReHLDS](https://github.com/dreamstalker/rehlds/releases/tag/3.13.0.788).

Yes, of course - it was me that suggested such feature after all.
The aforementioned feature works as intended, but it does not account for steam authentication, which seems to not work when connecting from a different game than the server (eg. connecting to a CS 1.6 server with a CS CZ client and vice-versa)

**Comment by anzz1** [comment]
I'm curious, why would someone want to join CS server with CS:CZ client?

I haven't had a deep look in the gamedlls of czero and compare it to cstrike but from my understanding there is little difference between the two.

So I guess it could work, but what I'm failing to understand is if the clients are 1:1, why bother. And if they are not, what is the advantage of using czero client over regular cstrike ?

A feature that allows players to join using a different game sounds like a recipe for trouble and should probably be disabled by default. Currently it looks like it's always enabled with no ability to disable without recompiling.

**Comment by SmilexGamer** [comment]
> I'm curious, why would someone want to join CS server with CS:CZ client?
> 
> I haven't had a deep look in the gamedlls of czero and compare it to cstrike but from my understanding there is little difference between the two.
> 
> So I guess it could work, but what I'm failing to understand is if the clients are 1:1, why bother. And if they are not, what is the advantage of using czero client over regular cstrike ?
> 
> A feature that allows players to join using a different game sounds like a recipe for trouble and should probably be disabled by default. Currently it looks like it's always enabled with no ability to disable without recompiling.

1. Just as a way to bridge the gap between both games. Originally, CS 1.6 and CS CZ players were supposed to be able to play together when CS 1.7 would eventually be released, but it was never done. So nowadays, both games can't play on the same servers.

2. They use the exact same gamedll, byte by byte.

3. Refer to point 1.

4. Yes, that's one of the flaws the current implmentation has: No way to disable the functionality and Steam authentication doesn't work for clients connecting from a game different from the server.

**Comment by jonatan1024** [comment]
This works if you disable steam authentication with `+sv_lan 1`.

Together with `+mp_consistency 0 +sv_net_incoming_decompression_max_ratio 100` I was able to connect to cstrike server with czero client and vice versa.

**Comment by SmilexGamer** [comment]
> This works if you disable steam authentication with `+sv_lan 1`.
> 
> Together with `+mp_consistency 0 +sv_net_incoming_decompression_max_ratio 100` I was able to connect to cstrike server with czero client and vice versa.

Of course it works when Steam authentication is disabled. This issue was opened in hopes of having the Steam authentication working even when you connect with a game client different from the server.

**Comment by myosotis10** [comment]
> Did you tried on your CZ client to type: setinfo _gd cstrike From [this](https://github.com/dreamstalker/rehlds/commit/0c5ce53666811c4243ed5aa0c755dab26d2283d4) change, and by using [latest ReHLDS](https://github.com/dreamstalker/rehlds/releases/tag/3.13.0.788).

can you tell me where should i place this item please??

**Comment by SmilexGamer** [comment]
> > Did you tried on your CZ client to type: setinfo _gd cstrike From [this](https://github.com/dreamstalker/rehlds/commit/0c5ce53666811c4243ed5aa0c755dab26d2283d4) change, and by using [latest ReHLDS](https://github.com/dreamstalker/rehlds/releases/tag/3.13.0.788).
> 
> can you tell me where should i place this item please??

In your game console.

**Comment by jonatan1024** [comment]
@SmilexGamer This turned out harder than expected, but I've managed to come up with a fix. It doesn't really fit into rehlds, so I made it a module insted. Please check out [CrossAuth](https://github.com/jonatan1024/CrossAuth) and let me know whether it works for you

**Comment by SmilexGamer** [comment]
> @SmilexGamer This turned out harder than expected, but I've managed to come up with a fix. It doesn't really fit into rehlds, so I made it a module insted. Please check out [CrossAuth](https://github.com/jonatan1024/CrossAuth) and let me know whether it works for you

Interesting, though this could be incorporated in rehlds, right?
